.github/workflows/e2e-k8s-1.28.yaml

```diff
@@ -0,0 +1,106 @@
+name: E2E-K8S-1.24
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
+  pull_request: {}
+  workflow_dispatch: {}
+
+env:
+  # Common versions
+  GO_VERSION: '1.20'
+  KIND_ACTION_VERSION: 'v1.5.0'
+  KIND_VERSION: 'v0.20.0'
```

Comment:
perhaps upgrade to 0.21.0 here or in the later PR, since the v0.20.0 mainly supports up to k8s 1.27.
see https://github.com/kubernetes-sigs/kind/releases